stages/org.osbuild.xorrisofs

```diff
@@ -6,7 +6,89 @@ import sys
 import osbuild.api
 
 
-def main(inputs, output_dir, options):
+# Create a grub2 bootable iso
+def grub2(inputs, output_dir, options):
+    tree = inputs["tree"]["path"]
+    boot = options.get("boot", {})
+    filename = options["filename"]
+    vol_id = options["volid"]
+    sys_id = options.get("sysid")
+    efi = options.get("efi")
+    isolevel = options.get("isolevel")
+    grub2 = options.get("grub2")
+
+    cmd = [
```

Comment:
There are a lot of identical options passed for this and the syslinux version of the function, i..e rock/joliet/-iso-level/-sysid for a start, probably boot-info-table,boot-laod-size,boot-catalog-hide and the target trees. I wonder if we could extract a common helper? I guess it's currently difficult because we have no (quick) test for the helper, I did a tiny and very primitve smoke test in https://github.com/osbuild/osbuild/compare/main...mvo5:main-grub2-xorrisofs-test?expand=1 which might make refactoring this easier, ideally of course it would create a real image not doing this mocking.

This is of course fine for a followup and I am happy to look at it if you would prefer that.

Reply:
I fiddled around with something like that at first, I think it just makes the code harder to read. Part of the problem is that the order of the arguments to xorrisofs is important, for some of them, so it's much easier to read and debug with everything in one place.